From 1ad1240fcb7c0bf4ca86d3a3ff250405f37e84bd Mon Sep 17 00:00:00 2001
From: Peter Maydell <peter.maydell@linaro.org>
Date: Mon, 6 Aug 2012 16:50:19 +0100
Subject: [PATCH 70/77] kvm-all.c: Move init of irqchip_inject_ioctl out of
 kvm_irqchip_create()

Move the init of the irqchip_inject_ioctl field of KVMState out of
kvm_irqchip_create() and into kvm_init(), so that kvm_set_irq()
can be used even when no irqchip is created (for architectures
that support async interrupt notification even without an in
kernel irqchip).

Signed-off-by: Peter Maydell <peter.maydell@linaro.org>
---
 kvm-all.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kvm-all.c b/kvm-all.c
index 34b02c1..b713c03 100644
--- a/kvm-all.c
+++ b/kvm-all.c
@@ -1200,7 +1200,6 @@ static int kvm_irqchip_create(KVMState *s)
         return ret;
     }
 
-    s->irqchip_inject_ioctl = KVM_IRQ_LINE;
     if (kvm_check_extension(s, KVM_CAP_IRQ_INJECT_STATUS)) {
         s->irqchip_inject_ioctl = KVM_IRQ_LINE_STATUS;
     }
@@ -1350,6 +1349,8 @@ int kvm_init(void)
     s->direct_msi = (kvm_check_extension(s, KVM_CAP_SIGNAL_MSI) > 0);
 #endif
 
+    s->irqchip_inject_ioctl = KVM_IRQ_LINE;
+
     ret = kvm_arch_init(s);
     if (ret < 0) {
         goto err;
-- 
1.7.9.5

